llvm/lib/Transforms/IPO/RepoPruning.cpp

```diff
@@ -314,9 +315,39 @@ bool RepoPruning::runOnModule(Module &M) {
     } while (IsEliminated);
   }
 
+  // Enable this pass support for alias.
+  for (Module::alias_iterator I = M.alias_begin(), E = M.alias_end(); I != E;) {
+    Module::alias_iterator CurI = I;
+    ++I;
+    // TODO: An Alias doesn't have an aliasee, for example:
+    // @v = alias i32, inttoptr(i32 42 to i32*)
+    if (auto *Aliasee = CurI->getBaseObject()) {
+      if (wasPruned(*Aliasee)) {
+        Type *Ty = CurI->getValueType();
+        // Delete the alias from Module if its aliasee is pruned.
+        CurI->removeFromParent();
+        // Replace the alias by a declaration.
+        llvm::Value *Declaration;
```

Comment:
A small point, but it would be better if we could avoid having this uninitialized variable. Could we break the code that creates the declaration (basically the `if` statement below) into a function so that you can write `Declaration=createDeclaration(...)`?

Reply:
Good point. I have changed the code following your advice. Thanks.